types/node/util.d.ts

```diff
@@ -1105,6 +1105,211 @@ declare module 'util' {
+    This is technically incorrect but is a much nicer UX for the common case.
+    The IfDefaultsTrue version is for things which default to true; the IfDefaultsFalse version is for things which default to false.
+    They differ only in the final item of the if-else chain.
+    */
+    type IfDefaultsTrue<T, IfTrue, IfFalse> = T extends true
+        ? IfTrue
+        : T extends false
+        ? IfFalse
+        : T extends boolean
+        ? IfTrue | IfFalse
+        : IfTrue;
+
+    type IfDefaultsFalse<T, IfTrue, IfFalse> = T extends true
+        ? IfTrue
+        : T extends false
+        ? IfFalse
+        : T extends boolean
+        ? IfTrue | IfFalse
+        : IfFalse;
+
+    type ExtractOptionValue<T extends ParseArgsConfig, O extends ParseArgsOptionConfig> = IfDefaultsTrue<
+        T['strict'],
+        O['type'] extends 'string' ? string : O['type'] extends 'boolean' ? boolean : string | boolean,
+        string | boolean
+    >;
+
+    type ParsedValues<T extends ParseArgsConfig> = (T['options'] extends ParseArgsOptionsConfig
+        ? {
+              -readonly [LongOption in keyof T['options']]: IfDefaultsFalse<
+                  T['options'][LongOption]['multiple'],
+                  undefined | Array<ExtractOptionValue<T, T['options'][LongOption]>>,
+                  undefined | ExtractOptionValue<T, T['options'][LongOption]>
+              >;
+          }
+        : {}) &
+        IfDefaultsTrue<T['strict'], unknown, { [longOption: string]: undefined | string | boolean }>;
+
+    type ParsedPositionals<T extends ParseArgsConfig> = IfDefaultsTrue<
+        T['strict'],
+        IfDefaultsFalse<T['allowPositionals'], string[], []>,
+        IfDefaultsTrue<T['allowPositionals'], string[], []>
+    >;
+
+    type PreciseTokenForOptions<
+        T extends ParseArgsConfig,
+        K extends keyof T['options'] & string,
+        O extends ParseArgsOptionConfig,
+    > = O['type'] extends 'string'
+        ? {
+              kind: 'option';
+              index: number;
+              name: K;
+              rawName: string;
+              value: string;
+              inlineValue: boolean;
+          }
+        : O['type'] extends 'boolean'
+        ? {
+              kind: 'option';
+              index: number;
+              name: K;
+              rawName: string;
+              value: undefined;
+              inlineValue: undefined;
+          }
+        : OptionToken & { name: K };
+
+    type TokenForOptions<
+        T extends ParseArgsConfig,
+        K extends keyof T['options'] = keyof T['options'],
+    > = K extends unknown
+        ? T['options'] extends ParseArgsOptionsConfig
+            ? PreciseTokenForOptions<T, K & string, T['options'][K]>
+            : OptionToken
+        : never;
+
+    type ParsedOptionsToken<T extends ParseArgsConfig> = IfDefaultsTrue<T['strict'], TokenForOptions<T>, OptionToken>;
+
+    type ParsedPositionalToken<T extends ParseArgsConfig> = IfDefaultsTrue<
```

Comment:
```suggestion
    type ParsedPositionalsToken<T extends ParseArgsConfig> = IfDefaultsTrue<
```

Light suggestion to match the pluralization of `ParsedOptionsToken` above

Reply:
Nice catch, fixed (though I changed `ParsedOptionsToken` instead.)